compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -1123,6 +1194,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             return tcx.ty_error();
         }
 
+        if regular_traits.iter().filter(|t| t.trait_ref().bound_vars().len() > 0).count() > 1 {
```

Comment:
So, I think this can be removed now. We shouldn't normally have multiple regular trait refs, but can under error conditions.